tests/test_url_build.py

```diff
@@ -32,12 +32,16 @@ def test_build_with_scheme_and_host():
     assert u == URL("http://127.0.0.1")
 
 
-def test_build_with_port():
-    with pytest.raises(ValueError):
-        URL.build(port=8000)
-
-    u = URL.build(scheme="http", host="127.0.0.1", port=8000)
-    assert str(u) == "http://127.0.0.1:8000"
+@pytest.mark.parametrize(
+    ("port", "exc", "match"),
+    [
+        pytest.param(8000, ValueError, None, id="port-only"),
+        pytest.param("", TypeError, "port is required to be int", id="port-str"),
```

Comment:
Let's use raw-strings for regexes:
```suggestion
        pytest.param("", TypeError, r"^port is required to be int$", id="port-str"),
```